scripts/pipeline/orthology_benchmark.py

```diff
@@ -17,18 +17,134 @@
 
 Typical usage example::
 
-    $ python orthology_benchmark.py --mlss_conf /path/to/mlss_conf.xml \
-        --species_set default
+    $ python orthology_benchmark.py --mlss_conf /path/to/mlss_conf.xml --species_set name \
+    --host mysql-ens-compara-prod-X --port XXXX --user ensro --out_dir /path/to/out/dir
 
 """
 
 import argparse
+import os
+import subprocess
+from typing import Dict, List
+
+import MySQLdb
+import numpy
 
 from ensembl.compara.config import get_species_set_by_name
 
 
-def dump_genomes():
-    """Docstring"""
+def dump_genomes(species_list: List[str], species_set_name: str, host: str, port: int,
+                 user: str, out_dir: str) -> None:
+    """Dumps canonical peptides of protein-coding genes for a specified list of species
+    from the latest available core databases to FASTA files using `dump_gene_set_from_core.pl`.
+
+    Args:
+        species_list: A list of species (genome) names to dump.
+        species_set_name: Species set (collection) name.
+        host: Database host.
+        port: Host port.
+        user: Server username.
+        out_dir: Directory to place `species_set_name/core_name.fasta` dumps.
+
+    Raises:
+        FileExistsError: If directory `out_dir/species_set_name` already exists.
+        MySQLdb.OperationalError: If connection to `host:port` cannot be established.
+        RuntimeError: If no cores are found for any species in `species_list` (`species_set_name`)
+                on `host:port`.
+
+    """
+    # Check access to the host & port
+    MySQLdb.connect(host=host, port=port, user=user)
+
+    cores = get_core_names(species_list, host)
+    dump_cores = [core for species, core in cores.items() if core != ""]
+
+    if len(dump_cores) == 0:
+        raise RuntimeError(f"No cores found for the species set '{species_set_name}' on host '{host}'.")
+
+    dumps_dir = os.path.join(out_dir, species_set_name)
+    os.mkdir(dumps_dir)
+
+    script = os.path.join(os.environ["ENSEMBL_ROOT_DIR"], "ensembl-compara", "scripts", "dumps",
+                          "dump_gene_set_from_core.pl")
+
+    for core in dump_cores:
+        out_file = os.path.join(dumps_dir, core + ".fasta")
+        subprocess.run([script, "-core-db", core, "-host", host, "-port", str(port),
+                       "-outfile", out_file], capture_output=True, check=True)
```

Comment:
Why `subprocess.run([...], capture_output=True, check=True)` instead of `subprocess.check_call([...])`?

Reply:
`subprocess.check_call` is part of an older high-level API (prior to Python 3.5). https://docs.python.org/3/library/subprocess.html#older-high-level-api

From 3.5 on, `subprocess.run` is a recommended way to invoke a subprocess:
> The recommended approach to invoking subprocesses is to use the run() function for all use cases it can handle. For more advanced use cases, the underlying Popen interface can be used directly.
>
> The run() function was added in Python 3.5; if you need to retain compatibility with older versions, see the Older high-level API section.

https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module

Reply:
indeed I also prefer to use run than check_call